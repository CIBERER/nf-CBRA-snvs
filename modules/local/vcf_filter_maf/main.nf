process VCF_FILTER_MAF {
    tag "${meta.id}"
    label 'process_single'

    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
      'https://depot.galaxyproject.org/singularity/ubuntu:20.04' :
      'nf-core/ubuntu:20.04' }"

    input:
    tuple val(meta), path(vep_tab)
    val maf

    output:
    tuple val(meta), path("${prefix}.maf.filtered.tab"), emit: maf_filtered_tab

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    prefix = task.ext.prefix ?: "${meta.id}"

    """
    zcat ${vep_tab} | awk -v thresh=${maf} '
    BEGIN { FS = OFS = "\\t"; header_found = 0 }
    # Keep all header lines
    /^#/ {
        print
        # Detect column header (single # at beginning)
        if (!header_found && \$0 ~ /^#[^#]/) {
            header_found = 1
            for (i=1; i<=NF; i++) {
                if (\$i == "MAX_AF") {
                    max_af_col = i
                    break
                }
            }
            if (!max_af_col) {
                print "ERROR: MAX_AF column not found." > "/dev/stderr"
                exit 1
            }
        }
        next
    }
    # Filter data rows based on MAX_AF threshold
    {
        af = \$max_af_col == "." || \$max_af_col == "" ? 0 : \$max_af_col
        if (af < thresh) print
    }
    ' > ${prefix}.maf.filtered.tab

    """

    stub:
    prefix = task.ext.prefix ?: "${meta.id}"
    """
    touch ${prefix}.maf.filtered.tab

    """
}