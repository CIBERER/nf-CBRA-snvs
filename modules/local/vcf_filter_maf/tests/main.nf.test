nextflow_workflow {

    name "Test Module vcf_filter_maf"
    script "../main.nf"
    workflow "VCF_FILTER_MAF"

    tag "modules"
    tag "modules/local/vcf_filter_maf"

    test("filter vep output by MAF 1 - tab ") {


        when {
            workflow {
                """
                input[0] = Channel.of([
                    [ id:'test'], // meta map
                    file("https://zenodo.org/records/15738776/files/NA12878-HG001.vep.tab.gz?download=1", checkIfExists: true)
                ])
                input[1] = Channel.value(1.01)
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success},
                { assert snapshot(workflow.out).match()}
            )
        }
    }


    test("filter vep output by MAF 0.001 - tab ") {


        when {
            workflow {
                """
                input[0] = Channel.of([
                    [ id:'test'], // meta map
                    file("https://zenodo.org/records/15738776/files/NA12878-HG001.vep.tab.gz?download=1", checkIfExists: true)
                ])
                input[1] = Channel.value(0.001)
                """
            }
        }

        then {
            assertAll(
                { assert workflow.success},
                { assert snapshot(workflow.out).match()}
            )
        }
    }

}
