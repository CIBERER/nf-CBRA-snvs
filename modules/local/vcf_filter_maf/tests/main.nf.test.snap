{
    "filter vep output by MAF 0.001 - tab ": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test.maf.filtered.tab:md5,0815261b2733582ff24ed54924e5c98d"
                    ]
                ],
                "maf_filtered_tab": [
                    [
                        {
                            "id": "test"
                        },
                        "test.maf.filtered.tab:md5,0815261b2733582ff24ed54924e5c98d"
                    ]
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.04.4"
        },
        "timestamp": "2025-07-09T14:34:47.466947476"
    },
    "filter vep output by MAF 1 - tab ": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test"
                        },
                        "test.maf.filtered.tab:md5,70c6f663d9c3d00e6d7d87042c6803c9"
                    ]
                ],
                "maf_filtered_tab": [
                    [
                        {
                            "id": "test"
                        },
                        "test.maf.filtered.tab:md5,70c6f663d9c3d00e6d7d87042c6803c9"
                    ]
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.04.4"
        },
        "timestamp": "2025-07-09T14:34:36.051925162"
    }
}